eservice/pdo/sservice/wsgi/store_blocks.py

```diff
@@ -71,13 +74,30 @@ def __call__(self, environ, start_response) :
             # block_list will be an iterator for blocks in the request, this prevents
             # the need to make a copy of the data blocks
             block_list = self.block_data_iterator(request)
-            raw_result = self.block_store.store_blocks(block_list, expiration=expiration, encoding='b64')
+            block_hashes = self.block_store.store_blocks(block_list, duration=duration, encoding='b64')
         except Exception as e :
             logger.exception('StoreBlocksApp')
             return ErrorResponse(start_response, "unknown exception while storing blocks")
 
         try :
-            result = json.dumps(raw_result).encode('utf8')
+            # going to just concatenate all hashes, safe since these are all fixed size
+            signing_hash_accumulator = duration.to_bytes(32, byteorder='big', signed=False)
+            signing_hash_accumulator += b''.join(block_hashes)
+
+            signing_hash = hashlib.sha256(signing_hash_accumulator).digest()
```

Comment:
Curious: Is there a reason we don't use the PDO crypto library's `compute_message_hash()` for this hash computation?

Reply:
Largely because it is symmetric.